node/tests/SharedTests.ts

```diff
@@ -3396,6 +3396,100 @@ export function runBaseTests<Context>(config: {
         },
         config.timeout,
     );
+
+    it.each([ProtocolVersion.RESP2, ProtocolVersion.RESP3])(
+        `lpos test_%p`,
+        async (protocol) => {
+            await runTest(async (client: BaseClient) => {
+                const key = `{key}:${uuidv4()}`;
+                const valueArray = ["a", "a", "b", "c", "a", "b"];
+                expect(await client.rpush(key, valueArray)).toEqual(6);
+
+                // simplest case
+                expect(await client.lpos(key, "a")).toEqual(0);
+                expect(await client.lpos(key, "b", { rank: 2 })).toEqual(5);
+
+                // element doesn't exist
+                expect(await client.lpos(key, "e")).toBeNull();
+
+                // reverse traversal
+                expect(await client.lpos(key, "b", { rank: -2 })).toEqual(2);
+
+                // unlimited comparisons
+                expect(await client.lpos(key, "a", { rank: 1, maxLength:0 })).toEqual(0);
+
+                // limited comparisons
+                expect(await client.lpos(key, "c", { rank: 1, maxLength:2 })).toBeNull();
+
+                // invalid rank value
+                try {
+                    expect(await client.lpos(key, "a", {rank: 0})).toThrow();
+                } catch (e) {
+                    expect((e as Error).message).toMatch(
```

Comment:
I'm not sure if its necessary to check the error message or not. Some tests do in this file, and some don't. Maybe we can decide as a team and align, or check with AWS what they prefer